framework/packages/abstract-sdk/src/base/endpoints/ibc_callback.rs

```diff
@@ -1,10 +1,11 @@
-use abstract_core::{ibc::IbcResponseMsg, IBC_CLIENT};
+use crate::core::AbstractError;
+use crate::{base::Handler, features::AbstractRegistryAccess, AbstractSdkError, ModuleInterface};
+use abstract_core::objects::module_reference::ModuleReference;
+use abstract_core::{ibc::IbcResponseMsg, objects::module::ModuleInfo, IBC_CLIENT};
 use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
 
-use crate::{base::Handler, AbstractSdkError, ModuleInterface};
-
 /// Trait for a contract's IBC callback ExecuteMsg variant.
-pub trait IbcCallbackEndpoint: Handler + ModuleInterface {
+pub trait IbcCallbackEndpoint: Handler + ModuleInterface + AbstractRegistryAccess {
```

Comment:
If we use VC then we can remove `ModuleInterface`. Although I would argue using `ModuleInterface` is better as it ensures that the account actually has IBC enabled.

Reply:
That doesn't work on adapters because there is no associated account (the account is found using the sender and not the contract storage). So what we can do is implement this version on adapters and keep the previous one on apps to keep this condition